sound/soc/sof/intel/hda-ipc.c

```diff
@@ -234,19 +230,20 @@ irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
 {
 	struct snd_sof_dev *sdev = context;
 	int ret = IRQ_NONE;
+	u32 irq_status = 0;
 
 	spin_lock(&sdev->hw_lock);
 
 	/* store status */
-	sdev->irq_status = snd_sof_dsp_read(sdev, HDA_DSP_BAR,
-					    HDA_DSP_REG_ADSPIS);
+	irq_status = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_REG_ADSPIS);
+	dev_vdbg(sdev->dev, "irq handler: irq_status:0x%x\n", irq_status);
 
 	/* invalid message ? */
-	if (sdev->irq_status == 0xffffffff)
+	if (irq_status == 0xffffffff)
 		goto out;
 
 	/* IPC message ? */
-	if (sdev->irq_status & HDA_DSP_ADSPIS_IPC) {
+	if (irq_status & HDA_DSP_ADSPIS_IPC) {
```

Comment:
I am confused. Can we have a race condition now, or is the commit message incorrect?

Reply:
@plbossart yes, there isnt a race anymore. Just to check for an IPC interrupt before waking up the irq thread. 
@keyonjie there is a typo in the commit message "wakeing" -> "waking?

Reply:
correct, there WAS race, but no anymore after the PR merged, do you think it is OK with "We are using" to state this status?

@ranj063 thanks, let me correct the typo, looks the spell check hook didn't help here.